app/models/manageiq/providers/openstack/network_manager/security_group.rb

```diff
@@ -55,6 +83,42 @@ def delete_security_group_queue(userid)
     MiqTask.generic_action_with_callback(task_opts, queue_opts)
   end
 
+  def delete_security_group_rule_queue(userid, key)
+    task_opts = {
+      :action => "delete Security Group rule for user #{userid}",
+      :userid => userid
+    }
+    queue_opts = {
+      :class_name  => self.class.name,
+      :method_name => 'raw_delete_security_group_rule',
+      :instance_id => id,
+      :priority    => MiqQueue::HIGH_PRIORITY,
+      :role        => 'ems_operations',
+      :zone        => ext_management_system.my_zone,
+      :args        => [key]
+    }
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
+  def raw_create_security_group_rule(security_group_id, direction, options)
+    options.delete_if { |_k, v| v.nil? || v.empty? }
```

Comment:
A nit, is the "array cleaning" needed for Fog? It doesn't look to be present in other _raw_create..._ or _raw_update..._ methods in this file.